ccm_web/client/src/utils/validation.ts

```diff
@@ -51,3 +62,21 @@ export function validateString (value: string | undefined, schema: StringSchema)
   }
   return { transformedValue, isValid, messages }
 }
+
+export function validateNumberString (value: string | undefined, schema: NumberSchema): ValidationResult<number> {
+  let isValid = true
+  let messages: string[] = []
+
+  const valueToTest = value === undefined ? value : value.trim() === '' ? NaN : Number(value)
```

Comment:
in case of `value` undefined or empty the code allows to go to next step which is `schema.validateSync(valueToTest, { strict: true })` 
1. what will happen when undefined is passed to `validateSync`?
2.  can the return statement in this scenario be done earlier, without passing through try-catch block.

Reply:
It could, but if you look at `createCanvasIdentifierSchema` factory function above I used to create `sectionIdSchema`, I'm setting `typeError` and `required`. These should be the messages when the value is `NaN` or `undefined`, respectively. I wanted `yup` to handle the creation of error messages, rather than pass more information into `validateNumberString`.

I know this looks somewhat convoluted. Basically my problem was I had to use `{ strict: true }`  because `yup` uses `parseFloat` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat) for value transformation. `parseFloat` has some weird behavior; basically it lops off non-numeric stuff at the end, where I wanted it to consider that an error. `Number` also has issues, since it treats an empty string as `0`. Hopefully this helps explain my decisions. I'm happy to consider alternatives if you have ideas.

Reply:
Thanks for explanation. We can keep it as is, I don't see a huge problem since this is not making any API call things as such, so the amount of time it takes stepping into that code in case of `undefined` and `NaN` is very little not worth our time now. This would be a case of keeping the code simple and readable.